### ISSUE TITLE
cy.intercept is having a long initial connection

### ISSUE THREAD

**Issue by kalimerre** [cypress-io/cypress#16234]
### Current behavior
Since we migrate from Cypress v6.9.1 to v7.X (we tested 7.1.0 & 7.2.0), we have noticed that sometimes some requests intercepted were taking a long time to execute (while response is being mocked and intercepted).
The issue is present in Chrome + Electron browser. 
In the following screenshots you'll see that most of the requests (to the same domain) which are intercepted works fine and somehow it just failed (because of the timeout) or is near to fail as it's done before the 5s timeout. 
The initial connection time seems to be random and never the same (from 1s to +5s).
This happens in cypress run and cypress open mode. 
![chrome](https://user-images.githubusercontent.com/5664180/116282296-dec96c80-a78a-11eb-8cf5-bd6163259a86.PNG)
![electron](https://user-images.githubusercontent.com/5664180/116282390-fb65a480-a78a-11eb-9e67-b95dec3b2e91.PNG)

### Desired behavior
cy.intercept is working the same way as it is working with Cypress v6.9.1 which means initial connection is at 0ms (or very low value) and that our tests are working fine without any timeout for this specific case. 

### Test code to reproduce
Still working on it to try to provide you something but it's quite complicated as it's totally random on Cypress v7+ so sometimes it'll works fine sometimes not ...

### Versions

Last known working Cypress version : v6.9.1
Cypress version where the issue is : v7.1.0 / v7.2.0
Browser : Chrome (90 on cypress open) / Electron (the one bundled)
It's the same in our Gitlab pipeline which is using the docker image base 14.16.0

**Comment by kalimerre** [comment]
Hello,

I was not able to provide a sample code or a repository which contains the issue (as I said it's random as sometimes it works sometimes not - mainly not working as our Gitlab pipeline which contains the v7 is never succeeding because of these timeout ...).

Nevertheless I'm able to provide you some debug logs.
I've used the DEBUG parameter : 

`DEBUG=cypress:network:*,cypress:net-stubbing*,cypress:server*,-cypress:server:plugins*,-cypress:server:tasks,-cypress:server:screenshot`

By doing this I was able to identify a common starting & ending point between v6 & v7, so I've extracted everything in-between.

The v6 is the version which is working fine without timeout.
[cypress_v6_clean.txt](https://github.com/cypress-io/cypress/files/6393179/cypress_v6_clean.txt)
The v7 is the one which is very slow to execute & intercept and could lead to timeout if it exceeds 4s of loading.
[cypress_v7_clean.txt](https://github.com/cypress-io/cypress/files/6393178/cypress_v7_clean.txt)

_I've replaced my real domain by mydomain.com and removed the query parameters from the GET request + mockedResponse for privacy reasons_

If you need me to do more testing or try to include more logs just tell me.

Thanks!

**Comment by kalimerre** [comment]
Hello,

Did anyone had a chance to look at this issue ? It's quite problematic for us and is blocking the update to v7 on our side so we have to stick with v6.9.1, but we would need the v7 for some new features which are not part of v6 ...

Thanks!

**Comment by jennifer-shehane** [comment]
It'd be helpful is we had a way to reproduce this to see it on our side. Is there a way you could reduce the example to something we can run and see ourselves in order to track down the issue?

**Comment by kalimerre** [comment]
@jennifer-shehane unfortunately I've tried to make anexample, but I was not able to provide a sample code which contains the issue. I think the only way is to get the whole codebase to reproduce it but even with it, it's not always happening on the same request / at the same point. That's why it's quite complicated to provide something ...

**Comment by kalimerre** [comment]
I have tested it with Cypress v7.3.0 and we still have the same issue in our pipeline renovate branch.

**Comment by flotwig** [comment]
> it's not always happening on the same request / at the same point. That's why it's quite complicated to provide something ...

Does it happen if you simply make a bunch of intercepted requests to https://jsonplaceholder.cypress.io/todos/1 in a loop?

**Comment by kalimerre** [comment]
I have tried with the following code but didn't notice any issues so far by running it ...
```
describe('mockedResponse', () => {
  beforeEach(() => {
    cy.intercept({
      method: 'GET',
      pathname: '/todos/1'
    }, (req) => {
      const body = 'hello'
      const headers = { 'x-custom-headers': 'mocked by cy.intercept' }
      req.reply(body, headers)
    }).as('mockedResponse')
  })
  for (let i = 0; i < 30; i++) {
    it('mockedResponse is rendered for the ' + i + ' times', () => {
      // start visiting the widget
      for (let j = 0; j < 30; j++) {
        cy.window().then(win => win.fetch('https://jsonplaceholder.cypress.io/todos/1'))
        cy.wait('@mockedResponse')
          .then((interception) => {
            cy.log(interception)
          })
          .its('response.headers')
          .should('have.property', 'x-custom-headers', 'mocked by cy.intercept')
      }
    })
  }
})
``` 

Can it be linked somehow to #16451 ?

**Comment by jennifer-shehane** [comment]
@kalimerre The fix for 16451 isn't release yet, but you could download a pre-release version with the fix by following instructions here https://github.com/cypress-io/cypress/commit/157e363a2da4d7ae7dcb22be39f1928f57c4382d#commitcomment-50685770

**Comment by agugut** [comment]
Hello @jennifer-shehane Do you have an estimated date to release the fix?

**Comment by jennifer-shehane** [comment]
@agugut We schedule a release every 2 weeks on Monday, so May 24 is the next scheduled release date.

**Comment by agugut** [comment]
> @agugut We schedule a release every 2 weeks on Monday, so May 24 is the next scheduled release date.

Thanks to answer me

**Comment by kalimerre** [comment]
Still identical with 7.4.0 which had just been released 😢

**Comment by eduardocalixtokorp** [comment]
> Still identical with 7.4.0 which had just been released 😢

Same here

**Comment by agugut** [comment]
Same here

**Comment by faileon** [comment]
Was this ever solved? We are running into same issue on 14.2.1, some mocked calls take 2000ms, while most are ~20ms. We haven't really found a pattern yet as to why is this happening and it appears random.
